syntax = "proto3";

package suggestions;

service SuggestionsService {
    rpc getSuggestions (getSuggestionsRequest) returns (SuggestionsResponse);
}

message VectorClock {
  map<string, int32> clock = 1;
}

message Book {
    string bookid = 1;
    string title = 2;
    string author = 3;
}

message getSuggestionsRequest{
    string bookid = 1;
    VectorClock vector_clock = 2;
}

message SuggestionsResponse {
    repeated Book items = 1;
    VectorClock vector_clock = 3;
}